src/common/file-structure.ts

```diff
@@ -58,9 +60,9 @@ export function buildRunFilePath({
   profileScope?: string;
   profileName: string;
   providerName: string;
-  language: 'JS' | 'Python';
+  language: SupportedLanguages;
 }): string {
-  const extension = language === 'JS' ? '.mjs' : '.py';
+  const extension = language === SupportedLanguages.JS ? '.mjs' : '.py';
```

Comment:
nitpick: When switching over the set of typed enums, I prefer to use maps typed something like `{ [key in SupportedLanguages]: string }`. That way when you extend `SupportedLanguages` (which is expected) the compiler will blow up in places that need to be addressed. This is especially useful when the various branching/switching is done in multiple places of the codebase.

So in this case it would look something like the following:

```ts
const EXTENSION_MAP: { [key in SupportedLanguages]: string } = {
  'js': '.mjs',
  'python': '.py'
}

const extension = EXTENSION_MAP[language]
```